Lesson4/src/com/home/RecruitingOffice.java

```diff
@@ -0,0 +1,80 @@
+package com.home;
+
+
+import com.home.UI.MyReader;
+import com.home.model.Address;
+import com.home.model.Person;
+
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+public class RecruitingOffice {
+    private Person[] recruits;
```

Comment:
this class needs to be refactored to work not with recruits, but with PersonRegistry

You need to delete all methods that are already defined in the PersonRegistry and call person registry to get all people by address, and then do verifications that you are doing in constructor

Reply:
may be you mean i must delete all methods in RecruitingOffice? if not, i'll not understand why need delete all.

Reply:
correct, you need to delete from the RecruitingOffice  methods that are already contained in the PersonRegistry